File: CF/B_Fireworks.py
from math import lcm
for _ in range(int(input())):
    a,b,c = map(int,input().split(" "))

    ans = 2
    tmp = lcm(a,b)
    c += tmp

    ans += c//a - tmp//a
    ans += c//b - tmp//b

    print(ans)

File: CF/B_Fireworks.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

long long gcd(long long a, long long b){
    return !b ? a : gcd(b, a%b);
}

long long lcm(long long a, long long b){
    return a*b/gcd(a,b);
}

int main(){
    int tt; cin >> tt;

    while(tt--){
        long long a,b,m; cin >> a >> b >> m;

        long long ans = 2;
        long long tmp = lcm(a,b);

        m += tmp;

        ans += m/a - tmp/a;
        ans += m/b - tmp/b;

        cout << ans << "\n";




    }
}